packages/app/src/templates/services/template.service.ts

```diff
@@ -122,26 +126,45 @@ export class TemplateService {
     }
   }
 
-  public async updateTemplateOrder(
-    template: Template
-  ): Promise<RequestResponse<Template>> {
-    const response = await this.deviceService.request({
-      endpoint: Endpoint.Messages,
-      method: Method.Put,
-      body: TemplatePresenter.mapToPureTemplateOrder(template),
+  public async updateTemplatesOrder(
+    templates: Template[]
+  ): Promise<UpdateTemplateOrderRequestResponse> {
+    const results = templates.map(async (template) => {
+      const { status } = await this.deviceService.request({
+        endpoint: Endpoint.Messages,
+        method: Method.Put,
+        body: TemplatePresenter.mapToPureTemplateOrder(template),
+      })
+      return {
+        status,
+        template,
+      }
     })
 
-    if (isResponseSuccess(response)) {
-      this.templateRepository.update(template)
+    const errorTemplates = (await Promise.all(results))
+      .filter(({ status }) => status === RequestResponseStatus.Error)
+      .map(({ template }) => template)
+    const successTemplates = (await Promise.all(results))
+      .filter(({ status }) => status === RequestResponseStatus.Ok)
+      .map(({ template }) => template)
+
+    if (errorTemplates.length > 0) {
+      successTemplates.forEach((template) =>
+        this.templateRepository.update(template)
```

Comment:
I think that it would be good to add test that checks how many times `templateRepository.update` was called and if the proper params are passed - this is the important part of this method.

Reply:
Test added in [43e06d1](https://github.com/mudita/mudita-center/pull/804/commits/43e06d134aad31f5e351a09965e5574fbf64604f)